### PR TITLE
Fix regression in jwt url parameter by awaiting async getAdditionalAuthHeader

### PR DESCRIPTION
**Pull request by cwperks**
### Description

A regression was introduced into JWT auth in the 2.4.0 release that prevented authentication with the JWT backend via the url param due to a signature change in the `getAdditionalAuthHeader` function. See this comment for details: https://github.com/opensearch-project/security-dashboards-plugin/issues/872#issuecomment-1372456283

This PR introduces a file called `after_env.js` which is needed to get beyond an issue `ReferenceError: TextEncoder is not defined` when trying to import `SignJWT` from jose.

The solution was suggested here: https://github.com/inrupt/solid-client-authn-js/issues/1676#issuecomment-917016646 

I'm not sure if this is the best way to get beyond the problem and open to anyone's suggestions. 

### Category
[Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation]

Bug Fix

### Issues Resolved

https://github.com/opensearch-project/security-dashboards-plugin/issues/872

### Testing

Added new selenium tests directly in this PR.

### Check List
- [X] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/jest_integration/jwt_auth.test.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      .setProtectedHeader({ alg: 'HS256' }) // algorithm
+      .setIssuedAt()
+      .sign(key);
+    const driver = getDriver(browser, options).build();
+    await driver.get(`http://localhost:5601/app/opensearch_dashboards_overview?token=${token}`);
+    await driver.wait(until.elementsLocated(By.xpath(pageTitleXPath)), 10000);
+
+    const cookie = await driver.manage().getCookies();
+    expect(cookie.length).toEqual(1);
+    await driver.manage().deleteAllCookies();
```
Comment (RyanL1997) [inline test/jest_integration/jwt_auth.test.ts:241]: Should we do an `afterEach` for those cookies clean up?

**Review by DarshitChanpura — APPROVED** [review]

File: test/jest_integration/jwt_auth.test.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    const token = await new SignJWT(payload) // details to  encode in the token
+      .setProtectedHeader({ alg: 'HS256' }) // algorithm
+      .setIssuedAt()
+      .sign(key);
+    const driver = getDriver(browser, options).build();
+    await driver.get(`http://localhost:5601/app/opensearch_dashboards_overview?token=${token}`);
+    await driver.wait(until.elementsLocated(By.xpath(pageTitleXPath)), 10000);
+
+    const cookie = await driver.manage().getCookies();
+    expect(cookie.length).toEqual(1);
```
Comment (RyanL1997) [inline test/jest_integration/jwt_auth.test.ts:240]: Can we also have a negative test case for a bad token?

File: test/jest_integration/jwt_auth.test.ts [hunk 3]
```diff
@@ -35,7 +35,9 @@ describe('start OpenSearch Dashboards server', () => {
   const pageTitleXPath = '//*[@id="osdOverviewPageHeader__title"]';
   // Browser Settings
   const browser = 'firefox';
-  const options = new Options().headless();
+  //const options = new Options().headless();
+  const options = new Options();
```
Comment (cwperks) [inline test/jest_integration/jwt_auth.test.ts:39]: Can you revert this back to headless when done?

Comment (RyanL1997) [reply test/jest_integration/jwt_auth.test.ts:39]: Nice catch! I'm updating that now.

Comment (RyanL1997) [reply test/jest_integration/jwt_auth.test.ts]: Done!

File: test/jest_integration/jwt_auth.test.ts (hunk 2 above)
Comment (RyanL1997) [reply test/jest_integration/jwt_auth.test.ts:240]: Updated with the negative test cases.

**Review by DarshitChanpura — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Review by cliu123 — APPROVED** [review]

**Pull request merged**